api/deploy.go

```diff
@@ -72,6 +47,31 @@ type DeployStatus struct {
 	Type    StatusType
 }
 
+// DeployServiceOption is a configuration func for deploying services.
+type DeployServiceOption func(*serviceDeployer)
+
+// DeployServiceStatusOption receives chan statuses to send deploy statuses.
+func DeployServiceStatusOption(statuses chan DeployStatus) DeployServiceOption {
+	return func(deployer *serviceDeployer) {
+		deployer.statuses = statuses
+	}
+}
+
+// DeployService deploys a service from a gzipped tarball.
+func (a *API) DeployService(r io.Reader, env map[string]string, options ...DeployServiceOption) (*service.Service,
+	*importer.ValidationError, error) {
+	return newServiceDeployer(a, env, options...).FromArchive(r)
+}
+
+// DeployServiceFromURL deploys a service living at a Git host.
+// Supported URL types:
+// - https://github.com/mesg-foundation/service-ethereum
+// - https://github.com/mesg-foundation/service-ethereum#branchName
+func (a *API) DeployServiceFromURL(url string, env map[string]string, options ...DeployServiceOption) (*service.Service,
+	*importer.ValidationError, error) {
+	return newServiceDeployer(a, env, options...).FromURL(url)
```

Comment:
Why not refactoring this one as well with just `DeployServiceFromURL` and `DeployService` that uses directly the content of `FromURL` and `FromArchive`?

Reply:
yeah, we can do that and keep the `newServiceDeployer` since it's used in both places

Reply:
done